packages/emitter-framework/src/python/components/function-declaration/function-declaration.test.tsx

```diff
@@ -1,23 +1,351 @@
+import { getOutput } from "#python/test-utils.jsx";
 import { Tester } from "#test/test-host.js";
+import * as py from "@alloy-js/python";
 import { t } from "@typespec/compiler/testing";
 import { describe, expect, it } from "vitest";
-import { getOutput } from "../../test-utils.js";
 import { FunctionDeclaration } from "./function-declaration.jsx";
 
-describe("Typescript Function Declaration", () => {
-  describe("Function bound to Typespec Types", () => {
-    describe("Bound to Operation", () => {
-      it("creates a function", async () => {
-        const { program, getName } = await Tester.compile(t.code`
-          op ${t.op("getName")}(id: string): string;
-        `);
-
-        expect(getOutput(program, [<FunctionDeclaration type={getName} />])).toRenderTo(`
-          def get_name(id: str) -> str:
-            pass
-          
-          `);
-      });
-    });
+describe("Python Function Declaration", () => {
+  it("creates a function", async () => {
+    const { program, getName } = await Tester.compile(t.code`
+      op ${t.op("getName")}(id: string): string;
+    `);
+
+    expect(getOutput(program, [<FunctionDeclaration type={getName} />])).toRenderTo(`
+      def get_name(id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates an async function", async () => {
+    const { program, getName } = await Tester.compile(t.code`
+      op ${t.op("getName")}(id: string): string;
+    `);
+
+    expect(getOutput(program, [<FunctionDeclaration async type={getName} />])).toRenderTo(`
+      async def get_name(id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates an async function with parametersModel", async () => {
+    const { program, createPerson, Foo } = await Tester.compile(t.code`
+      op ${t.op("createPerson")}(id: string): string;
+
+      model ${t.model("Foo")} {
+        name: string;
+        age: int32;
+      }
+    `);
+
+    expect(
+      getOutput(program, [<FunctionDeclaration async type={createPerson} parametersModel={Foo} />]),
+    ).toRenderTo(`
+      async def create_person(name: str, age: int) -> str:
+        pass
+      
+      `);
+  });
```

Comment:
What is the use case for this? Why wouldn't the TSP definition be
```typespec
op createPerson(...Foo): string;
```
in order to produce this Python function?

Reply:
That may indeed not be needed, but I added that to mimic this functionality that's available in the Typescript implementation: https://github.com/pinterest/typespec/blame/main/packages/emitter-framework/src/typescript/components/function-declaration.tsx#L71

I can remove it, if you think that's not needed.